hypershift-operator/controllers/nodepool/nodepool_controller_test.go

```diff
@@ -1476,3 +1477,187 @@ func Test_validateHCPayloadSupportsNodePoolCPUArch(t *testing.T) {
 		})
 	}
 }
+
+func TestSupportedVersionSkewCondition(t *testing.T) {
+	pullSecret, err := os.ReadFile("../../../hack/dev/fakePullSecret.json")
+	if err != nil {
+		t.Fatalf("failed to read manifests file: %v", err)
+	}
+	basePullSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pull-secret",
+			Namespace: "test-ns",
+		},
+		Data: map[string][]byte{
+			corev1.DockerConfigJsonKey: pullSecret,
+		},
+	}
+
+	baseNodePool := &hyperv1.NodePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-np",
+			Namespace: "test-ns",
+		},
+		Spec: hyperv1.NodePoolSpec{
+			Release: hyperv1.Release{
+				Image: "quay.io/openshift-release-dev/ocp-release:4.18.5-x86_64",
+			},
+		},
+		Status: hyperv1.NodePoolStatus{
+			Conditions: []hyperv1.NodePoolCondition{},
+		},
+	}
+
+	baseHostedCluster := &hyperv1.HostedCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-hc",
+			Namespace: "test-ns",
+		},
+		Spec: hyperv1.HostedClusterSpec{
+			PullSecret: corev1.LocalObjectReference{
+				Name: "pull-secret",
+			},
+		},
+		Status: hyperv1.HostedClusterStatus{
+			Version: &hyperv1.ClusterVersionStatus{
+				History: []configv1.UpdateHistory{
+					{
+						State:   configv1.CompletedUpdate,
+						Version: "4.18.5",
+					},
+				},
+			},
+		},
+	}
+
+	testCases := []struct {
+		name              string
+		nodePool          *hyperv1.NodePool
+		hostedCluster     *hyperv1.HostedCluster
+		releaseProvider   *fakereleaseprovider.FakeReleaseProvider
+		expectedCondition *hyperv1.NodePoolCondition
+		expectedError     string
+	}{
+		{
```

Comment:
can we also cover the case where if len(hcluster.Status.Version.History) == 0 {

Reply:
done